pilot/pkg/model/authentication.go

```diff
@@ -90,6 +90,23 @@ func initAuthenticationPolicies(env *Environment) (*AuthenticationPolicies, erro
 	return policy, nil
 }
 
+// MightAffectNamespaceByPeerAuthn determines whether it's possible to have a peer authentication policy
+// to affect the workloads in a particular namespace.
+func (policy *AuthenticationPolicies) MightAffectNamespaceByPeerAuthn(namespace string) bool {
```

Comment:
Suggest defining an interface for AuthenticationPolicies

Reply:
confused how willl that help? `cluster.go` code has reference of `pushContext.AuthenticationPolicies` and invoke this function directly. and that's the only calls of this method.

Reply:
I thought for each version, there is be an instance implementing this function.

Reply:
Not for this. The function is basically for a specific version (PeerAuthn)